broker/src/main/java/io/camunda/zeebe/broker/bootstrap/CommandApiServiceStep.java

```diff
@@ -0,0 +1,202 @@
+                          messagingService.bindingAddresses(),
+                          messagingService.address());
+
+                      concurrencyControl.run(
+                          () -> startServerTransport(brokerStartupContext, startupFuture));
+                    }
+                  },
+                  startupFuture);
+            });
+  }
+
+  @Override
+  void shutdownInternal(
+      final BrokerStartupContext brokerShutdownContext,
+      final ConcurrencyControl concurrencyControl,
+      final ActorFuture<BrokerStartupContext> shutdownFuture) {
+
+    final var commandApiServiceActor = brokerShutdownContext.getCommandApiService();
+    if (commandApiServiceActor == null) {
+      closeServerTransport(brokerShutdownContext, concurrencyControl, shutdownFuture);
+      return;
+    }
+    brokerShutdownContext.removePartitionListener(commandApiServiceActor);
+    brokerShutdownContext.removeDiskSpaceUsageListener(commandApiServiceActor);
+
+    concurrencyControl.runOnCompletion(
+        commandApiServiceActor.closeAsync(),
+        proceed(
+            () -> {
+              brokerShutdownContext.setCommandApiService(null);
+              closeServerTransport(brokerShutdownContext, concurrencyControl, shutdownFuture);
+            },
+            shutdownFuture));
+  }
+
+  private void startServerTransport(
+      final BrokerStartupContext brokerStartupContext,
+      final ActorFuture<BrokerStartupContext> startupFuture) {
+
+    final var concurrencyControl = brokerStartupContext.getConcurrencyControl();
+    final var brokerInfo = brokerStartupContext.getBrokerInfo();
+    final var schedulingService = brokerStartupContext.getActorSchedulingService();
+    final var messagingService = brokerStartupContext.getCommandApiMessagingService();
+
+    final var atomixServerTransport =
+        new AtomixServerTransport(brokerInfo.getNodeId(), messagingService);
+
+    concurrencyControl.runOnCompletion(
+        schedulingService.submitActor(atomixServerTransport),
+        proceed(
+            () -> {
+              brokerStartupContext.setCommandApiServerTransport(atomixServerTransport);
+              startCommandApiService(brokerStartupContext, startupFuture);
+            },
+            startupFuture));
+  }
+
+  private void startCommandApiService(
+      final BrokerStartupContext brokerStartupContext,
+      final ActorFuture<BrokerStartupContext> startupFuture) {
+
+    final var concurrencyControl = brokerStartupContext.getConcurrencyControl();
+    final var brokerInfo = brokerStartupContext.getBrokerInfo();
+    final var brokerCfg = brokerStartupContext.getBrokerConfig();
+    final var schedulingService = brokerStartupContext.getActorSchedulingService();
+    final var messagingService = brokerStartupContext.getCommandApiMessagingService();
+
+    final var transportFactory = new TransportFactory(schedulingService);
+
+    final var serverTransport =
```

Comment:
:question: This recreates an `AtomixServerTransport`, which was already created once in the `startServerTransport` method. Is this on purpose? Why are we creating two? Are we using the one we created beforehand?

Reply:
Nope, copy and paste error